synapseclient/client.py

```diff
@@ -3104,6 +3110,20 @@ def _findTeam(self, name):
         for result in self._GET_paginated("/teams?fragment=%s" % name):
             yield Team(**result)
 
+    def _find_teams_for_principal(
+        self, principal_id: str
+    ) -> typing.Generator[Team, None, None]:
```

Comment:
I am not using this in a way that requires a Generator, [further reading on the generator](https://docs.python.org/3.8/library/typing.html#typing.Generator). I am swapping this over to:
`def _find_teams_for_principal(self, principal_id: str) -> typing.Iterator[Team]` instead since it's a simple list iteration